plugin/hexmode.vim

```diff
@@ -101,29 +99,29 @@ if has("autocmd")
 
         " before writing a file when editing in hex mode, convert back to non-hex
         au BufWritePre *
-            \ if exists("b:editHex") && b:editHex && &binary |
-            \  let oldview = winsaveview() |
-            \  let oldro=&ro | let &ro=0 |
-            \  let oldma=&ma | let &ma=1 |
+            \ if exists("b:editHex") && b:editHex && &l:binary |
+            \  let b:oldview = winsaveview() |
+            \  let b:oldro=&l:ro | let &l:ro=0 |
+            \  let b:oldma=&l:ma | let &l:ma=1 |
             \  undojoin |
             \  silent exe "%!xxd -r" |
-            \  let &ma=oldma | let &ro=oldro |
-            \  unlet oldma | unlet oldro |
-            \  let &undolevels = &undolevels |
+            \  let &l:ma=b:oldma | let &l:ro=b:oldro |
+            \  unlet b:oldma | unlet b:oldro |
+            \  let &l:ul = &l:ul |
             \ endif
 
         " after writing a binary file, if we're in hex mode, restore hex mode
         au BufWritePost *
-            \ if exists("b:editHex") && b:editHex && &binary |
-            \  let oldro=&ro | let &ro=0 |
-            \  let oldma=&ma | let &ma=1 |
+            \ if exists("b:editHex") && b:editHex && &l:binary |
+            \  let b:oldro=&l:ro | let &l:ro=0 |
+            \  let b:oldma=&l:ma | let &l:ma=1 |
             \  undojoin |
             \  silent exe "%!xxd" |
-            \  exe "set nomod" |
-            \  let &ma=oldma | let &ro=oldro |
-            \  unlet oldma | unlet oldro |
-            \  call winrestview(oldview) |
-            \  let &undolevels = &undolevels |
+            \  exe "setlocal nomod" |
+            \  let &l:ma=:oldma | let &l:ro=b:oldro |
+            \  unlet b:oldma | unlet b:oldro |
+            \  call winrestview(b:oldview) |
+            \  let &l:ul = &l:ul |
```

Comment:
It seems that way, I completely agree, but it came along with documentation in #6.